src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/ReadMePreviewViewModel.cs

```diff
@@ -0,0 +1,112 @@
+{
+    public sealed class ReadMePreviewViewModel : ViewModelBase
+    {
+        public ReadMePreviewViewModel()
+        {
+            _isErrorWithReadMe = false;
+            _rawReadMe = string.Empty;
+        }
+
+        private bool _isErrorWithReadMe = false;
+
+        public bool IsErrorWithReadMe
+        {
+            get => _isErrorWithReadMe;
+            set
+            {
+                if (_isErrorWithReadMe != value)
+                {
+                    _isErrorWithReadMe = value;
+                    RaisePropertyChanged(nameof(IsErrorWithReadMe));
+                }
+            }
+        }
+
+        private string _rawReadMe = string.Empty;
+
+        public string ReadMeMarkdown
+        {
+            get => _rawReadMe;
+            set
+            {
+                if (_rawReadMe != value)
+                {
+                    _rawReadMe = value;
+                    RaisePropertyChanged(nameof(ReadMeMarkdown));
+                }
+            }
+        }
+
+        private bool _canDetermineReadMeDefined = true;
+
+        public bool CanDetermineReadMeDefined
+        {
+            get => _canDetermineReadMeDefined;
+            set
+            {
+                if (_canDetermineReadMeDefined != value)
+                {
+                    _canDetermineReadMeDefined = value;
+                    RaisePropertyChanged(nameof(CanDetermineReadMeDefined));
+                }
+            }
+        }
+
+        public async Task LoadReadme(string packagePath, string packageId)
+        {
+            var newReadMeValue = string.Empty;
+            var isErrorWithReadMe = false;
+            bool canDetermineReadMeDefined = false;
+#pragma warning disable CA1031 // Do not catch general exception types
+            try
+            {
+                if (!string.IsNullOrEmpty(packagePath))
+                {
+                    var packageDirectory = Path.GetDirectoryName(packagePath);
+                    var nuspecPath = Path.Combine(packageDirectory, $"{packageId}{PackagingCoreConstants.NuspecExtension}");
+                    var nuspectFileInfo = new FileInfo(nuspecPath);
+                    if (nuspectFileInfo.Exists)
+                    {
+                        canDetermineReadMeDefined = true;
+                        var nuspecReader = new NuspecReader(nuspecPath);
+                        var readMePath = nuspecReader.GetReadme();
+                        if (!string.IsNullOrEmpty(readMePath))
+                        {
+                            var readMeFullPath = Path.Combine(packageDirectory, readMePath);
+                            var readMeFileInfo = new FileInfo(readMeFullPath);
+                            if (readMeFileInfo.Exists)
+                            {
+                                using var readMeStreamReader = readMeFileInfo.OpenText();
+                                var readMeContents = await readMeStreamReader.ReadToEndAsync();
```

Comment:
Please make sure that this happens on a background thread, and not on the UI thread.  Maybe throw in a `ThreadHelper.ThrowIfOnUIThread()`

Reply:
Per [rule #1 of JTF](https://github.com/Microsoft/vs-threading/blob/main/doc/threading_rules.md#rule-1-if-a-method-has-certain-thread-apartment-requirements-sta-or-mta-it-must-either) threading, In async code, you should do await TaskScheduler.Default; 

Probably a good idea to explicitly switch anyways.

Reply:
I made a change to switch to the default thread and then back to the UI thread when updating the properties.